### ISSUE TITLE
Ardour hangs after stopping playback

### ISSUE THREAD

**Issue by paulfd** [sfztools/sfizz#549]
Ardour hangs for a while after stopping playback. The duration of that freeze seems to be proportional to the amount of sfizz instruments that are used.

This is a regression that was introduced with 846dd6d2 "Load a default SFZ if path empty". On 0.5.1 (currently 1ebed26b), after reverting 846dd6d2 (and fixing the resulting conflicts), the problem disappears and sfizz appears to work correctly (no issues after working on a project for some hours).

Operating system: Gentoo Linux

**Steps to reproduce:**

1. Open a new session in Ardour 6.3.0 (either with ALSA or jack backend)
2. Create a MIDI track with sfizz as instrument and add a note in it
3. Start and stop playback, all seems okay
4. Create 3 more MIDI tracks (adding notes is not needed)
5. Start and stop playback, the meters and the whole UI freeze for a while
6. Create 10 more MIDI tracks
7. Start and stop playback, the meters and the whole UI freeze for a long time (7 seconds)

Selecting sfz files in the sfizz plugin does not seem to affect it.

Reported by midgard on IRC, they remain available for troubleshooting.

**Comment by ghost** [comment]
I'm not sure whether this is the same issue but Ardour 6.3 and the whole system has graphical freezes independent of whether notes are added or more MIDI tracks are added. Soon after loading sfizz, Ardour (and I believe the whole system) exhibits graphical window freezes that last several seconds or more. 

O/S: Linux Mint XFCE
DAW: Ardour 6.3

**Comment by jpcima** [comment]
This might be fixed in the latest, do you mind to try it again?

**Comment by paulfd** [comment]
No freezes on start/stop playback for me on Ardour 6.5 with the develop branch. On the other hand, instantiating 10 sfizz tracks takes longer than what I'd find comfortable, but once instantiation is done it seems OK.

**Comment by ghost** [comment]
No freezes here now for me on dev branch.

**Comment by paulfd** [comment]
Let's assume this is solved then, we can reopen if necessary.

**Comment by lminiero** [comment]
I was about to open a new issue, until I found an existing one already. It actually still happens to me, any time I hit stop. Apparently it started after I recently upgraded my Fedora installation to 34, which now ships Ardour 6.8. It don't remember this happening before, with sfizz 0.4.0 and Ardour 6.3, while the same sfizz version started to cause the issue, which caused me to wonder whether the issue was in sfizz or Ardour. Upgrading sfizz to the 1.0.0 release version didn't fix the issue for me, and neither did installing the latest `develop` branch.

I can confirm that it seems to be related to how many sfizz instances are loaded. In my case, a larger orchestral template locks the UI for about 20 seconds, as you can see from this log excerpt (10.57.11 is about when I stopped playback):

```
2021-09-12T10:57:11 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e3c640860
2021-09-12T10:57:12 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e3b5b58a0
2021-09-12T10:57:12 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2d71eb90
2021-09-12T10:57:13 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e33679880
2021-09-12T10:57:14 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2c9c2cd0
2021-09-12T10:57:15 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e30c82cd0
2021-09-12T10:57:16 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e29d48350
2021-09-12T10:57:17 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e384efd00
2021-09-12T10:57:17 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e357f89d0
2021-09-12T10:57:18 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2bbb13f0
2021-09-12T10:57:19 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2e7b9350
2021-09-12T10:57:20 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e346af610
2021-09-12T10:57:21 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e3a864920
2021-09-12T10:57:22 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2f25ab20
2021-09-12T10:57:22 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e36839790
2021-09-12T10:57:23 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e39a524e0
2021-09-12T10:57:24 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e303ebd50
2021-09-12T10:57:25 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e38f92510
2021-09-12T10:57:26 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2fbd0a90
2021-09-12T10:57:26 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2b0f1e00
2021-09-12T10:57:27 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e31dcf0a0
2021-09-12T10:57:28 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e2a7cea90
2021-09-12T10:57:29 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e3149fe90
2021-09-12T10:57:30 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e37baadd0
2021-09-12T10:57:30 [INFO]: Removing MIDI patch file custom:Sfizz LV2:0x564e32738e10
```

Disabling all plugins and enabling one at a time for testing showed how the issue starts to become noticeable after ~3/4 instances active at the same time.

**Comment by lminiero** [comment]
Just double checking whether you want me to open a new issue for this, and in general if there's any additional info I can provide. Thanks!
